packages/lab/src/tooltip/Tooltip.tsx

```diff
@@ -1,131 +1,143 @@
+   * Whether to hide the state icon within the tooltip. Defaults to `false`.
    */
   hideIcon?: boolean;
   /**
-   * A callback function to render the tooltip content
-   * @param {function} getIcon getter for the icon based on the state
-   * @param {function} getIconProps getter for the icon properties
+   * Content displayed inside the tooltip. Can be a string or a React component.
    */
-  render?: (props: TooltipRenderProp) => ReactNode;
+  content: ReactNode;
   /**
    * A string to determine the current status of the tooltip
    */
-  status?: TooltipStatus;
-  title?: string;
+  status?: ValidationStatus;
+  /**
+   * Delay in milliseconds before the tooltip is shown
+   */
+  enterDelay?: number;
+  /**
+   * Delay in milliseconds before the tooltip is hidden
+   */
+  leaveDelay?: number;
+  /**
+   * Option to not display the tooltip. Can be used in conditional situations like text truncation.
+   */
+  disabled?: boolean;
   /**
-   * This prop is used to help implement the accessibility logic.
-   * If you don't provide this prop. It falls back to a randomly generated id.
+   * Option to remove the hover listener
    */
-  id?: string;
-  open?: boolean;
+  disableHoverListener?: boolean;
+  /**
+   * Option to remove the focus listener
+   */
+  disableFocusListener?: boolean;
 }
 
 export const Tooltip = forwardRef<HTMLDivElement, TooltipProps>(
-  function Tooltip(
-    {
-      arrowProps,
+  function Tooltip(props, ref) {
+    const {
       children,
-      className: classNameProp,
+      className,
       container,
+      disabled,
       disablePortal,
-      hideArrow,
-      hideIcon,
-      open,
-      render,
+      hideArrow = false,
+      hideIcon = false,
+      open: openProp,
+      content,
       status = "info",
-      title,
+      placement = "right",
+      enterDelay = 300,
+      leaveDelay = 0,
       ...rest
-    },
-    ref
-  ) {
-    const getIcon = useCallback(
-      (iconProps: IconProps) => {
-        if (hideIcon) {
-          return null;
-        }
-        return status !== "info" ? (
-          <StatusIndicator
-            status={status}
-            {...iconProps}
-            className={withBaseName("icon")}
-          />
-        ) : null;
-      },
-      [status, hideIcon]
-    );
+    } = props;
 
-    const Window = useWindow();
+    const hookProps: UseTooltipProps = {
+      open: openProp,
+      placement,
+      enterDelay,
+      leaveDelay,
+      ...rest,
+    };
 
-    if (!open) {
-      return null;
-    }
+    const { arrowProps, open, getTriggerProps, getTooltipProps } =
+      useTooltip(hookProps);
+
+    const { ref: tooltipRef, ...restTooltipProps } = getTooltipProps();
+
+    const { ref: triggerRefHook, ...restTriggerProps } = getTriggerProps();
+
+    const triggerRefMerged = useForkRef(
+      // @ts-ignore
+      isValidElement(children) ? children.ref : null,
+      triggerRefHook as Ref<HTMLDivElement>
```

Comment:
Do you think it's better to change useTooltip to return the floating and reference separately to the getters.

Or maybe this is an advanced case where the getters would look cleaner in a more typical use-case?

Reply:
Should look better now. Thanks